pkg/scheduler/scheduler.go

```diff
@@ -316,6 +316,11 @@ func New(client clientset.Interface,
 	}
 
 	addAllEventHandlers(sched, informerFactory, dynInformerFactory, unionedGVKs(clusterEventMap))
+
+	if dynInformerFactory != nil {
```

Comment:
@Huang-Wei why were we building this informer factory within `scheduler.New`, whereas the other factory is built and started in `server.go`?

Reply:
That _may_ work. In that case, we will have to instantiate a placeholder dynInformerFactory via Setup() (in cmd/.../config.go#Config), and pass it to scheduler.New(), and then start it in Run(). If it sounds good, I can raise a PR after this PR gets merged.

Reply:
I think that would be good for consistency.